### DISCUSSION TITLE
Visualize Privacy Manifests

### DISCUSSION THREAD

**Discussion by Jeehut** [SwiftPackageIndex/SwiftPackageIndex-Server#2423] [Ideas]
I'm sure you've already had thoughts about this, but I just wanted to raise this topic in case you have not.

Apple this year at WWDC introduced a new feature for SDK authors called "Privacy Manifests", see [these notes](https://wwdcnotes.com/notes/wwdc23/10060/) or [the session](https://developer.apple.com/videos/play/wwdc2023/10060).

I immediately thought of this project when I saw the video and what I would imagine is a small view summarizing some relevant aspects of the Privacy Manifests that might be important to decide as a developer if we want to include a specific library or not, similar to how the License is also important and presented.

What are your thoughts about this?

**Comment by finestructure** [comment]
This is definitely very interesting and I can totally see us surfacing this kind of information prominently on the package page 👍

**Comment by daveverwer** [comment]
It's something we've talked about before this feature existed, and I'm so glad to see Apple tackle it officially! 🎉

I've not looked into them in detail yet but will watch the session video this week and start thinking about the best way to display the data.

Is it something you'd be interested in helping contribute?

**Reply by Jeehut** [reply]
I don't really have time to do any open-source work recently, or else I would be live-streaming that part. So I don't want to promise anything. But when the topic comes up later this year again, feel free to ping me, then I can tell you how things are looking at the time.

**Reply by daveverwer** [reply]
No need to do anything, just checking if it was something you were wanting to do. 👍

**Comment by Jeehut** [comment]
@daveverwer @finestructure Just listened to your latest podcast episode where you also mentioned this discussion. One of the things you weren't sure was how to show off the information from the manifest files, so I thought I would share my expectations.

The most interesting two bits of info I would like to know are:
- What level of data tracking does this library involve, one of: "Data Not Collected", "Data Not Linked to You", "Data Linked to You"
- Is the library listed in Apples list of "Privacy-impacting SDKs"? If yes, I would expect a clear ❌ / ⚠️ style warning.

My expectation is driven by how Apple groups those labels [here](https://www.apple.com/privacy/labels/):
<img width="388" alt="Screenshot 2023-06-15 at 19 38 51" src="https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/assets/6942160/67dd0adc-feca-411e-907a-41e40fec7c70">
<img width="759" alt="Screenshot 2023-06-15 at 19 38 46" src="https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/assets/6942160/111cb106-984d-42b3-aaa3-5c803e24c0f5">

P.S. I'm used to this so no worries at all, but just for future reference, "Jeehut" is how you say my first name "Cihat" correctly. 😉

**Reply by daveverwer** [reply]
Yea, these are the "three panels" I referred to in the podcast, but there's a lot of information for an already quite busy page. The challenge will be to try and summarise it in a way where it does not dominate the page.

> P.S. I'm used to this so no worries at all, but just for future reference, "Jeehut" is how you say my first name "Cihat" correctly. 😉

Thanks for this, and, my apologies!

**Reply by Jeehut** [reply]
I‘m not sure if you’ve got my point. I don’t want all the information from these 3 groups. I just want to know the “max” group. So it could be just an icon or a word that signifies that. Like “None”, “Unlinked”, “Linked”.

**Reply by Sherlouk** [reply]
Don't take it personally if Dave gets your name wrong, you can know him for years and he'll still find a way to mess it up 🤣🤣

**Reply by daveverwer** [reply]
> Don't take it personally if Dave gets your name wrong, you can know him for years and he'll still find a way to mess it up 🤣🤣

Yes yes, you're never going to let me live that down, are you? 😂

The ironic thing is with "Cihat", I knew I was going to say it and so I found one of your YouTube videos where you said your name and even practiced saying it in the way you did as I prepared for the podcast, but recording something you know you are nervous about into a microphone always makes it harder! I did my research, but my execution failed 😂
